src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+import java.util.Scanner;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static final String DIR_NAME = "data";
+    private static final String FILE_NAME = "duke.txt";
+
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList tasks;
+
+    private int getIndex(String args) {
+        return Integer.parseInt(args) - 1;
+    }
+
+    /**
+     * Adds a new Todo to the TaskList.
+     *
+     * @param command Command with details about the Todo to be created.
+     * @throws IOException If an error occurs while writing to the file.
+     */
+    public void addTodo(Command command) throws IOException {
+        Task todo = new Todo(command.getArgs());
+        tasks.add(todo);
+        storage.writeToFile(todo);
+        ui.addedTaskReply(todo, tasks.size());
+    }
+
+    /**
+     * Adds a new Deadline to the TaskList
+     *
+     * @param command Command with details about the Deadline to be created.
+     * @throws IOException If an error occurs while writing to the file.
+     * @throws DateTimeParseException If date provided is not of the "dd/mm/yyyy hhmm" or "dd/mm/yyyy" format.
+     */
+    public void addDeadline(Command command) throws IOException, DateTimeParseException {
+        String[] parts = command.getArgs().split("/by");
+
+        if (parts.length != 2) {
+            throw new InvalidTaskException(command.getInstruction());
+        } else {
+            Task deadline = new Deadline(parts[0], parts[1]);
+            tasks.add(deadline);
+            storage.writeToFile(deadline);
+            ui.addedTaskReply(deadline, tasks.size());
+        }
+    }
+
+    /**
+     * Adds a new Event to the TaskList
+     *
+     * @param command Command with details about the Event to be created.
+     * @throws IOException If an error occurs while writing to the file.
+     */
+    public void addEvent(Command command) throws IOException {
+        String[] parts = command.getArgs().split("/at");
+
+        if (parts.length != 2) {
+            throw new InvalidTaskException(command.getInstruction());
+        } else {
+            Task event = new Event(parts[0], parts[1]);
+            tasks.add(event);
+            storage.writeToFile(event);
+            ui.addedTaskReply(event, tasks.size());
+        }
+    }
+
+    /**
+     * Marks a task as done
+     *
+     * @param command Command containing the index of the task to mark as done.
+     * @throws IOException If an error occurs while writing to the file.
+     */
+    public void done(Command command) throws IOException {
```

Comment:
Since this function is an action, would it be better to name this is a way that sounds like an action? (Eg: markAsDone())?